File: 1034.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int main() {
    int T;
    scanf("%d", &T);  // Número de instâncias
    while (T > 0) {
        int N, M;
        scanf("%d %d", &N, &M);  // Lê N e M
        int *blocos = (int *)malloc(N * sizeof(int));  // Lê os blocos disponíveis
        for (int i = 0; i < N; i++) {
            scanf("%d", &blocos[i]);
        }

        int *dp = (int *)malloc((M + 1) * sizeof(int));  // Inicializa dp
        for (int i = 0; i <= M; i++) {
            dp[i] = INT_MAX;  // Inicializa com infinito (valor alto)
        }
        dp[0] = 0;  // Base: comprimento 0 requer 0 blocos

        for (int i = 0; i < N; i++) {  // Iterando pelos tipos de blocos
            int bloco = blocos[i];
            for (int j = bloco; j <= M; j++) {  // Atualizando dp[j] para todos os valores possíveis
                if (dp[j - bloco] != INT_MAX) {
                    dp[j] = (dp[j] < dp[j - bloco] + 1) ? dp[j] : (dp[j - bloco] + 1);
                }
            }
        }

        printf("%d\n", dp[M] != INT_MAX ? dp[M] : -1);    
        free(blocos);
        free(dp);
        T--;  // Decrementa o número de instâncias
    }
    return 0;
}

File: 1034.py
import sys
T = int(sys.stdin.readline().strip())  # Número de instâncias
while T > 0:
    N, M = map(int, sys.stdin.readline().split())  # Lê N e M
    blocos = list(map(int, sys.stdin.readline().split()))  # Lê os blocos disponíveis

    dp = [float('inf')] * (M + 1)  # Inicializa com infinito (valor alto)
    dp[0] = 0  # Base: comprimento 0 requer 0 blocos

    i = 0
    while i < N:  # Iterando pelos tipos de blocos
        bloco = blocos[i]
        j = bloco
        while j <= M:  # Atualizando dp[j] para todos os valores possíveis
            dp[j] = min(dp[j], dp[j - bloco] + 1)
            j += 1
        i += 1

    print(dp[M] if dp[M] != float('inf') else -1 )    
    T -= 1  # Decrementa o número de instâncias
